test/integration/test_backend.py

```diff
@@ -198,6 +198,23 @@ def test_retrieve_backend_not_exist(self):
         with self.assertRaises(QiskitBackendNotFoundError):
             self.dependencies.provider.get_backend("nonexistent_backend")
 
+    def test_dynamic_circuits_warning(self):
+        """Test warning when user defines dynamic==False and circuits are dynamic"""
+        backend = self.dependencies.provider.get_backend("ibmq_qasm_simulator")
```

Comment:
The reason is that when I use a simulator/fake backend, then we get the warning ""Parameter 'dynamic' is False, but the circuit contains dynamic constructs." instead of the warning I am trying to hit, and therefore the test fails.